avm/src/daemon/ctrl.rs

```diff
@@ -112,21 +117,23 @@ impl LogRendezvousHash {
   // Runs a binary search for the record whose hash is closest to the key hash without
   // going over. If none are found, the *last* record in the list is returned as it wraps around.
   fn get_idx_for_key(&self, key: &str) -> usize {
+    let l = self.sorted_hashes.len();
     let mut key_hasher = XxHash64::with_seed(0xa1a2);
     key_hasher.write(key.as_bytes());
-    let key_hash = key_hasher.finish();
+    let key_hash = key_hasher.finish() % (l as u64);
```

Comment:
why does need to be done?

Reply:
It was my fast workaround for a bug below; it's not selecting the index based on the `key_hash`, it was accidentally treating the `key_hash` as the index, so I just made sure it was within the range of the indexes.

Basically, the range of indexes is just `0` and `1` in my test, while on average a hash of type `u64` would have the value of `2^63` and the odds of actually ever selecting `0` in the code were `1 : 2^64` which is way below the threshold of likelihood physicists use for quantum mechanics experiments.

I will undo this after I figure out how to make the rest of the function do what it should be doing.

Reply:
I think this is why the code was not working: https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search_by

> If the value is found then Result::Ok is returned, containing the index of the matching element. If there are multiple matches, then any one of the matches could be returned. If the value is not found then Result::Err is returned, containing the index where a matching element could be inserted while maintaining sorted order.

We're almost never going to find the value. The `Err` is not actually an error here, but a weird flag indicating which index we could insert this value into the vector (which indicates which node we want it to go to). I *think* I have an idea on how to resolve this, going to test.